lib/modules/home/home_presenter.dart

```diff
@@ -4,4 +4,77 @@ abstract class HomePresenter
     extends Presenter<HomeView, HomeInteractor, HomeRouter> {}
 
 class HomePresenterImpl extends HomePresenter
-    implements HomeViewDelegate, HomeInteractorDelegate {}
+    implements HomeViewDelegate, HomeInteractorDelegate {
+  HomePresenterImpl() {
+    stateDidInit.voidListen(_stateDidInit).addTo(disposeBag);
+    showDetailButtonDidTap.listen(_showDetailButtonDidTap).addTo(disposeBag);
+    didSwipeDown.voidListen(_didSwipeDown).addTo(disposeBag);
+
+    surveysDidFetch.listen(_surveysDidFetch).addTo(disposeBag);
+    surveysDidFailToFetch.listen(_surveysDidFailToFetch).addTo(disposeBag);
+    isSurveysCachedDidGet.listen(_isSurveysCachedDidGet).addTo(disposeBag);
+    authenticatedUserDidGet.listen(_authenticatedUserDidGet).addTo(disposeBag);
+  }
+
+  @override
+  final stateDidInit = BehaviorSubject<void>();
+
+  @override
+  final showDetailButtonDidTap = BehaviorSubject<SurveyInfo>();
+
+  @override
+  final didSwipeDown = BehaviorSubject<void>();
+
+  @override
+  final alertDialogDidClose = BehaviorSubject<void>();
+
+  @override
+  final surveysDidFailToFetch = BehaviorSubject<Exception>();
+
+  @override
+  final surveysDidFetch = BehaviorSubject<List<SurveyInfo>>();
+
+  @override
+  final isSurveysCachedDidGet = BehaviorSubject<bool>();
+
+  @override
+  final authenticatedUserDidGet = BehaviorSubject<UserInfo>();
+
+  void _stateDidInit() {
+    interactor.getIsSurveysCached();
+    interactor.getAuthenticatedUser();
+    view.setDateTime(DateTime.now());
+  }
+
+  void _showDetailButtonDidTap(SurveyInfo survey) {
+    router.pushToSurveyDetail(context: view.context, survey: survey);
+  }
+
+  void _surveysDidFetch(List<SurveyInfo> surveys) {
+    view.stopSkeletonAnimation();
+    view.dismissProgressHUD();
+    view.showSurveys(surveys);
+  }
+
+  void _surveysDidFailToFetch(Exception exception) {
+    view.stopSkeletonAnimation();
+    view.dismissProgressHUD();
+    view.alert(exception);
+  }
+
+  void _isSurveysCachedDidGet(bool isCached) {
+    interactor.fetchSurveys();
+    if (!isCached) {
+      view.beginSkeletonAnimation();
+    }
+  }
+
+  void _authenticatedUserDidGet(UserInfo user) {
+    view.showUser(user);
+  }
+
+  void _didSwipeDown() {
+    view.showProgressHUD();
+    interactor.fetchSurveys(force: true);
+  }
```

Comment:
imo, we should show pull to refresh indicator when swiping down to re-fetch surveys rather than showing your `ProgressHUD` 🙏 It's normal behavior on other apps as well.

Reply:
Besides, there is weird splash animation when I swipe down, could you please check? 😅 

https://user-images.githubusercontent.com/60863885/120165163-7a844780-c225-11eb-9749-9643207142a8.mp4

Reply:
@manh-t I found this bug and fixed this in another [PR](https://github.com/markgravity/flutter-survey/blob/fac8ce972635d296c0dde43045ebdf392347ce1a/lib/components/common/progress_hud.dart#L39)

Reply:
@manh-t 
> imo, we should show pull to refresh indicator when swiping down to re-fetch surveys rather than showing your ProgressHUD 🙏 It's normal behavior on other apps as well.

Let me update it in the same PR with paging feature

Reply:
> @manh-t I found this bug and fixed this in another [PR](https://github.com/markgravity/flutter-survey/blob/fac8ce972635d296c0dde43045ebdf392347ce1a/lib/components/common/progress_hud.dart#L39)

okay, I will recheck this issue later 🙆‍♂️